src/basic/cgroup-util.c

```diff
@@ -1688,6 +1688,36 @@ int cg_slice_to_path(const char *unit, char **ret) {
         return 0;
 }
 
+static int cg_is_threaded_one(const char *controller, const char *path) {
+        _cleanup_free_ char *fs = NULL, *contents = NULL;
+        int r;
+
+        r = cg_get_path(controller, path, "cgroup.type", &fs);
+        if (r < 0)
+                return r;
+
+        r = read_one_line_file(fs, &contents);
+        if (r == -ENOENT)
+                return false;
+        if (r < 0)
+                return r;
+
+        return streq_ptr(contents, "domain threaded");
+}
+
+int cg_is_threaded(const char *controller, const char *path) {
+        char prefix[strlen(path) + 1];
+        int r;
+
+        PATH_FOREACH_PREFIX_MORE(prefix, path) {
```

Comment:
hmm, the way I understood it regular cgroups can never be below threaded ones, hence going up the tree should not be necessary?